src/widgets/index/index.ts

```diff
@@ -450,7 +450,10 @@ const index = (widgetParams: IndexWidgetParams): IndexWidget => {
       };
 
       derivedHelper = mainHelper.derive(() =>
-        mergeSearchParameters(...resolveSearchParameters(this))
+        mergeSearchParameters(
+          mainHelper.state,
```

Comment:
this part here is to ensure that "statically set parameters" on `mainHelper` persist (parameters on `helper` don't, as that actually is the helper of the mainIndex, confusing naming, I know).

If we're doing such a refactor, we probably should simply get rid of `helper`, and replace it with `mainHelper` only (which then gets name swapped)